public/app/features/explore/state/actions.ts

```diff
@@ -185,83 +222,35 @@ export function initializeExplore(
   ui: ExploreUIState
 ): ThunkResult<void> {
   return async dispatch => {
-    const exploreDatasources: DataSourceSelectItem[] = getDatasourceSrv()
-      .getExternal()
-      .map(ds => ({
-        value: ds.name,
-        name: ds.name,
-        meta: ds.meta,
-      }));
-
+    dispatch(loadExploreDatasourcesAndSetDatasource(exploreId, datasourceName));
     dispatch(
       initializeExploreAction({
         exploreId,
         containerWidth,
         eventBridge,
-        exploreDatasources,
         queries,
         range,
         ui,
       })
     );
-
-    if (exploreDatasources.length >= 1) {
-      let instance;
-
-      if (datasourceName) {
```

Comment:
These lines have been somewhat refactored into `changeDatasource` don't really understand the `!instance` part so would appreciate any feedback on the refactor. 
https://github.com/grafana/grafana/blob/2ae63e70c0df3ce2e1cf478d4173be72428c3dd9/public/app/features/explore/state/actions.ts#L211-L230

Reply:
can not understand it either

Reply:
If the requested datasource is not available, it gets the default datasource (by not providing an argument to `get`).

Reply:
of course, was the comment about it checking if data source was delete that tripped me :)